src/qw/docsection.py

```diff
@@ -0,0 +1,708 @@
+"""
+DocX outline level iterator.
+
+Layer over python-docx allowing looping through
+outline sections, duplicating them as necessary
+to put the required data in.
+"""
+import re
+from copy import deepcopy
+from typing import Optional, TypeAlias
+
+import docx
+from docx.opc.constants import RELATIONSHIP_TYPE as RT
+from loguru import logger
+from lxml.etree import QName
+
+from qw import md
+from qw.base import QwError
+
+
+def qname(ns: str, val: str) -> str:
+    """Qualified name for "ns:val"."""
+    return QName(docx.oxml.ns.nsmap[ns], val)
+
+
+def nsw(val: str) -> str:
```

Comment:
might be worth having expanding out this function name so its a bit more readable?